src/Logary.Prometheus/Exporter.fs

```diff
@@ -0,0 +1,171 @@
+  //  # HELP http_request_duration_seconds A histogram of the request duration.
+  //  # TYPE http_request_duration_seconds histogram
+  //  http_request_duration_seconds_bucket{le="0.05"} 24054
+  //  http_request_duration_seconds_bucket{le="0.1"} 33444
+  //  http_request_duration_seconds_bucket{le="0.2"} 100392
+  //  http_request_duration_seconds_bucket{le="0.5"} 129389
+  //  http_request_duration_seconds_bucket{le="1"} 133988
+  //  http_request_duration_seconds_bucket{le="+Inf"} 144320
+  //  http_request_duration_seconds_sum 53423
+  //  http_request_duration_seconds_count 144320
+  let exportMetricInfo (sb: StringBuilder) nameTransform (metricInfo: BasicInfo * seq<MetricInfo>) =
+    let basicInfo, metricInfos = metricInfo
+    let metricName = nameTransform basicInfo.name
+    let help = escapeLabelValue basicInfo.description
+    let metricType = metricInfos |> Seq.first |> Option.map getTypeStr
+    if metricType.IsSome then
+      appendLine sb (sprintf "# HELP %s %s" metricName help)
+      appendLine sb (sprintf "# TYPE %s %s" metricName metricType.Value)
+      for metricInfo in metricInfos do
+        match metricInfo with
+        | Gauge gauge ->
+          let identifier = getIdentifier nameTransform metricName gauge.labels
+          let gaugeValueStr = gauge.gaugeValue |> formatFloat
+          appendLine sb (sprintf "%s %s" identifier gaugeValueStr)
+        | Histogram histogram ->
+          let mutable cumulativeCount = 0.
+          for (KeyValue(upperBound, count)) in histogram.bucketsInfo do
+            let labelsWithLe = histogram.labels |> Map.add "le" (upperBound |> formatFloat)
+            let bucketIdentifier = getIdentifier nameTransform (sprintf "%s_bucket" metricName) labelsWithLe
+            cumulativeCount <- cumulativeCount + count
+            appendLine sb (sprintf "%s %s" bucketIdentifier (cumulativeCount |> formatFloat))
+
+          let sumIdentifier = getIdentifier nameTransform (sprintf "%s_sum" metricName) histogram.labels
+          appendLine sb (sprintf "%s %s" sumIdentifier (histogram.sumInfo |> formatFloat))
+
+          let countIdentifier = getIdentifier nameTransform (sprintf "%s_count" metricName) histogram.labels
+          appendLine sb (sprintf "%s %s" countIdentifier (cumulativeCount |> formatFloat))
+
+
+  let exportToPrometheus exportConf =
+    let metricRegistry = exportConf.metricRegistry
+    let metricNameTransformer = exportConf.metricNameTransformer >> ExporterConf.nameValidator
+    let metricInfos = metricRegistry.getMetrictInfos ()
+    let sb = new System.Text.StringBuilder ()
+    metricInfos |> Seq.iter (exportMetricInfo sb metricNameTransformer)
+    sb.ToString()
+
+
+  let internal exportWebPart exportConf : WebPart =
+    let urlPath = exportConf.urlPath
+    path urlPath >=> GET  >=> setHeader "Content-Type" "text/plain; charset=utf-8" >=> request (fun r ->  Successful.OK (exportToPrometheus exportConf))
+
+
+  let runAsync token conf  =
```

Comment:
Why do we run it as a Task? Also, start as Task doesn't match up with the name of this function.

Reply:
you are right, 👍